extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1031,11 +1031,42 @@ export class Project implements ISqlProject {
 		const databaseReferenceEntry = new DacpacReferenceProjectEntry(settings);
 
 		// check if reference to this database already exists
+		// if it does, throw an error that will get displayed to the user
 		if (this.databaseReferenceExists(databaseReferenceEntry)) {
 			throw new Error(constants.databaseReferenceAlreadyExists);
 		}
 
-		await this.addToProjFile(databaseReferenceEntry);
+
+		const sqlProjectsService = await utils.getSqlProjectsService();
+
+		// create database variable
+		if (settings.databaseVariable && settings.databaseName) {
+			await sqlProjectsService.addSqlCmdVariable(this.projectFilePath, settings.databaseVariable, settings.databaseName);
+		}
+
+		// create server variable
+		if (settings.serverVariable && settings.serverName) {
+			await sqlProjectsService.addSqlCmdVariable(this.projectFilePath, settings.serverVariable, settings.serverName);
+		}
+
+		// need to set these to undefined if they are empty strings
+		const databaseVariable = settings.databaseVariable ? settings.databaseVariable : undefined;
```

Comment:
Right now STS only checks if these are null, which is why I had to update these here for the logic to work correctly. I'll send out a separate PR to change the STS check to check for null or whitespace instead of just null

Reply:
Added validation in STS in https://github.com/microsoft/sqltoolsservice/pull/1894, but as discussed in chat, I also updated the addDatabaseReferenceDialog to convert empty string values to undefined to make it more clear when these values were not set.